### PR TITLE
Fix #308 by refactoring most of AudioContext into BaseAudioContext.

### LINKED ISSUE #308: Remove unsupported methods from offlineAudioContext
The methods 

`createMediaElementSource`
`createMediaStreamSource`
`createMediaStreamDestination`

don't make sense in an `offlineAudioContext`. They should be removed from the spec. Robert O'Callahan [suggests](http://lists.w3.org/Archives/Public/public-audio/2014JanMar/0089.html) adding a `baseAudioContext` or similar to the spec which `AudioContext` and `offlineAudioContext` can inherit from. That seems sensible to me. 

[more details](http://lists.w3.org/Archives/Public/public-audio/2014JanMar/0088.html)

### PR DESCRIPTION
**Pull request by joeberkovitz**
AudioContext and OfflineAudioContext now inherit from BaseAudioContext.

### CODE REVIEW

**Comment by joeberkovitz** [comment]
@rtoy @padenot I am blocked on #250 by this PR because I don't want to introduce tons of new references to AudioContext only to have them invalidated by this change to BaseAudioContext.

File: index.html [hunk 1]
```diff
@@ -826,13 +833,13 @@ <h2 id="AudioContext">
           </dt>
           <dd>
             Closes the audio context, releasing any system audio resources used
-            by the <a>AudioContext</a>. This will not automatically release all
-            AudioContext-created objects, unless other references have been
+            by the <a>BaseAudioContext</a>. This will not automatically release
+            all AudioContext-created objects, unless other references have been
```
Comment (rtoy) [inline index.html]: "AudioContext-created" -> "BaseAudioContext-created"?

File: index.html [hunk 2]
```diff
@@ -4566,11 +4590,18 @@ <h2 id="AudioListener">
           The AudioListener Interface
         </h2>
         <p>
-          This interface interface is DEPRECATED, as it will be replaced by the
+          This interface represents the position and orientation of the person
```
Comment (rtoy) [inline index.html]: Hard to tell from the diff, but did you really want to get rid of the deprecation?

**Comment by joeberkovitz** [comment]
Thanks, these things are cleaned up now.

**Pull request merged**